File: Lab9/app/src/main/java/com/example/lab9/ElemViewHolder.java
package com.example.lab9;

import android.view.View;
import android.widget.TextView;

import androidx.recyclerview.widget.RecyclerView;

public class ElemViewHolder extends RecyclerView.ViewHolder {
    private TextView contentItemView;
    public String data;
    public long id;

    public ElemViewHolder(View contentItemView, String data) {
        super(contentItemView);
        this.contentItemView = (TextView) contentItemView.findViewById(R.id.item_text_view);
        this.data = data;
    }

    public void updateElem(long id, String data) {
        String textViewContent = data + " (" + String.valueOf(id) +")";
        contentItemView.setText(textViewContent);
        this.id = id;
        this.data = data;
    }

    public void setData(String data) {
        contentItemView.setText(data);
        this.data = data;
    }

    public void setId(long id) {
        this.id = id;
    }

    public long getId() {
        return id;
    }
}

File: Lab7/app/src/main/java/lab7/ElemViewHolder.kt
package com.example.koshelev.labs.javalab7

import android.view.View
import android.widget.TextView

import androidx.recyclerview.widget.RecyclerView

class ElemViewHolder(contentItemView: View, var data: String) : RecyclerView.ViewHolder(contentItemView) {
    private val contentItemView: TextView

    init {
        this.contentItemView = contentItemView.findViewById<View>(R.id.item_text_view) as TextView
    }

    fun setDataAndUpdate(data: String) {
        contentItemView.text = data
        this.data = data
    }
}
